lib/pagerduty/maintenance_client.rb

```diff
@@ -34,7 +36,19 @@ def get_raw(options = {})
         'filter' => 'open',
         'service_ids' => PagerDuty::Configuration.service_ids
       }.merge(options)
-      perform(:get, 'maintenance_windows', query).body
+
+      begin
+        perform(:get, 'maintenance_windows', query).body
+      rescue => e
```

Comment:
Does the error have a status?

Something like `if e.respond_to?(:status) && e.status == 400` could work if so

Reply:
yes. Thanks for calling this out. I had changed `conn.response` in [configuration.rb](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/pagerduty/configuration.rb#L14) to NOT return a custom error while I was debugging. So pd was returning the string, "the server responded with status 400." BUT, I changed it back so we're still returning the custom error, but I didn't account for that here! Let me change this up a bit. For the record, this is what the error is when there is a bad service ID:

on `master`:
`irb(main):001> PagerDuty::PollMaintenanceWindows.new.perform
2024-11-19 10:47:10.042220 E [66391:11540 base.rb:115] Rails -- BackendServiceException: {:status=>400, :detail=>nil, :code=>"VA900", :source=>nil} -- { :title => "Operation failed", :detail => "Operation failed", :code => "VA900", :status => "400" }`

Reply:
I put a `binding.pry` right here and did some digging. Based on this info, I'm going to use `e.status_code`
```
[1] pry(#<PagerDuty::MaintenanceClient>)> e
=> #<PagerDuty::ServiceException: PagerDuty::ServiceException>
[2] pry(#<PagerDuty::MaintenanceClient>)> e.message
=> "BackendServiceException: {:status=>400, :detail=>nil, :code=>\"VA900\", :source=>nil}"
[3] pry(#<PagerDuty::MaintenanceClient>)> e.message.include?('400')
=> true
[4] pry(#<PagerDuty::MaintenanceClient>)> e.status
NoMethodError: private method `status' called for an instance of PagerDuty::ServiceException
from (pry):4:in `rescue in get_raw'
Caused by PagerDuty::ServiceException: PagerDuty::ServiceException
from /Users/rebeccatolmach/git/vets-api/lib/common/client/base.rb:115:in `rescue in request'
Caused by Common::Exceptions::BackendServiceException: Common::Exceptions::BackendServiceException
from /Users/rebeccatolmach/git/vets-api/lib/common/client/middleware/response/raise_custom_error.rb:30:in `raise_error!'
[5] pry(#<PagerDuty::MaintenanceClient>)> e.exception
=> #<PagerDuty::ServiceException: PagerDuty::ServiceException>
[6] pry(#<PagerDuty::MaintenanceClient>)> e.respond_to?(:status)
=> false
[7] pry(#<PagerDuty::MaintenanceClient>)> e.status_code
=> 400
[8] pry(#<PagerDuty::MaintenanceClient>)> e.errors
=> [#<Common::Exceptions::SerializableError:0x00000003102926c0 @code="VA900", @detail=nil, @href=nil, @id=nil, @links=[], @meta=nil, @source=nil, @status="400", @title="Operation failed">]
[9] pry(#<PagerDuty::MaintenanceClient>)> e.exception
=> #<PagerDuty::ServiceException: PagerDuty::ServiceException>
[10] pry(#<PagerDuty::MaintenanceClient>)> e.respond_to?(:status_code)
=> true
```

Reply:
`e.status_code` won't work because a 401 _status_ returns a 400 _status code_ 😑 
binding.pry with a bad token (401):
```
[2] pry(#<PagerDuty::MaintenanceClient>)> e.message
=> "BackendServiceException: {:status=>401, :detail=>nil, :code=>\"VA900\", :source=>nil}"
[4] pry(#<PagerDuty::MaintenanceClient>)> e.status_code
=> 400
[6] pry(#<PagerDuty::MaintenanceClient>)> e.message.include?("401")
=> true
[10] pry(#<PagerDuty::MaintenanceClient>)> e.original_status
=> 401
```
I could go back to `.include` or I'll try `original_status`.